### DISCUSSION TITLE
Type hierarchy (class inheritance tree) support

### DISCUSSION THREAD

**Discussion by tdimitrov** [zed-industries/zed#16348] [Feature Requests]
To set up the stage - I'm a new Zed user test driving the editor with a C++ project.

VSCode has got a `Show type hierarchy' command. When a type is selected and you invoke it it generates a nice tree-like view with all the types inheriting or inherited by the selected one. Here are some screenshots.
Invoking:
![image](https://github.com/user-attachments/assets/6850f398-917f-43af-84bf-01d498bce2cf)
The result:
![image](https://github.com/user-attachments/assets/75e2bf4d-d0fa-429d-9ba9-d8bef07ab7b5)

Is there something similar to this in Zed? I tried looking through PRs, issues, discussions but maybe I'm not using the right keywords or the feature has got a different name.

**Comment by yeskunall** [comment]
Hey @tdimitrov (and for others that come across this discussion): we have a PR ready here that will add support for this feature: https://github.com/zed-industries/zed/pull/43750.

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/community-links
